[quant][graphmode] Test for per channel graph mode quantization

jerryzh168 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #29494 [quant][graphmode] Returning axis from calculate_qparams
* **#29493 [quant][graphmode] Test for per channel graph mode quantization**
* #29416 [quant] Make PerChannelMinMaxObserver scriptable using `torch.jit.ignore`

Summary:
att

Test Plan:
python test/test_jit.py

Reviewers:
mvz

Subscribers:

Tasks:

Tags:

Differential Revision: [D18580907](https://our.internmc.facebook.com/intern/diff/D18580907)

jerryzh168 committed:
[quant][graphmode] Test for per channel graph mode quantization

Summary:
att

Test Plan:
python test/test_jit.py

Reviewers:
mvz

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode] Test for per channel graph mode quantization"


Summary:
att

Test Plan:
python test/test_jit.py

Reviewers:
mvz

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode] Test for per channel graph mode quantization"


Summary:
att

Test Plan:
python test/test_jit.py

Reviewers:
mvz

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode] Test for per channel graph mode quantization"


Summary:
att

Test Plan:
python test/test_jit.py

Reviewers:
mvz

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode] Test for per channel graph mode quantization"


Summary:
att

Test Plan:
python test/test_jit.py

Reviewers:
mvz

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

ZolotukhinM commented on test/test_jit.py:
```diff
@@ -1106,8 +1120,10 @@ def forward(self, x):
-    def test_insert_quant_dequant(self):
+    @given(
+        is_per_channel=st.booleans()
+    )
```
Can we replace it with `for is_per_channel in [True, False]` and not pull hypothesis here?

jerryzh168 replied:
isn't this cleaner?

ZolotukhinM commented on test/test_jit.py:
```diff
@@ -1449,76 +1466,83 @@ def forward(self, x):
+        is_per_channel=st.booleans()
+    )
+    def test_fold_prepack(self, is_per_channel):
+        for name, M, ref_M, data in [
```
Having several ifs on the same condition is not great. How about we split this to several tests rather than iterating through different configurations in a loop in one test and extract the common part to a separate function?

jerryzh168 replied:
most of the code are the same, how about put the if conditions in small functions?

ZolotukhinM commented on test/test_jit.py:
```diff
@@ -36,10 +36,21 @@
 import torch.nn.functional as F
 from torch.quantization import QConfig
 from torch.quantization._quantize_script import ConvPackedParams, LinearPackedParams
+from torch.quantization._quantize_script import script_qconfig
```
We probably need to move all quantization related tests from this file into a separate file (no need to do it in this PR).

jerryzh168 committed:
Update on "[quant][graphmode] Test for per channel graph mode quantization"


Summary:
att

Test Plan:
python test/test_jit.py

Reviewers:
mvz

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode] Test for per channel graph mode quantization"


Summary:
att

Test Plan:
python test/test_jit.py

Reviewers:
mvz

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 commented:
moved some if statements to separate functions, please take a look again

jerryzh168 committed:
Update on "[quant][graphmode] Test for per channel graph mode quantization"


Summary:
att

Test Plan:
python test/test_jit.py

Reviewers:
mvz

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

ZolotukhinM reviewed (approved):
Thanks, looks good!

jerryzh168 committed:
Update on "[quant][graphmode] Test for per channel graph mode quantization"


Summary:
att

Test Plan:
python test/test_jit.py

Reviewers:
mvz

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode] Test for per channel graph mode quantization"


Summary:
att

Test Plan:
python test/test_jit.py

Reviewers:
mvz

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode] Test for per channel graph mode quantization"


Summary:
att

Test Plan:
python test/test_jit.py

Reviewers:
mvz

Subscribers:

Tasks:

Tags:

Differential Revision: [D18580907](https://our.internmc.facebook.com/intern/diff/D18580907)

[ghstack-poisoned]

jerryzh168 commented:
@pytorchbot retest this please

jerryzh168 committed:
Update on "[quant][graphmode] Test for per channel graph mode quantization"


Summary:
att

Test Plan:
python test/test_jit.py

Reviewers:
mvz

Subscribers:

Tasks:

Tags:

Differential Revision: [D18580907](https://our.internmc.facebook.com/intern/diff/D18580907)

[ghstack-poisoned]

jamesr66a commented on test/test_jit.py:
```diff
@@ -36,10 +36,21 @@
+from torch.quantization import quantize
+from common_quantization import SingleLayerLinearModel, AnnotatedSingleLayerLinearModel
+from common_quantization import ConvModel, AnnotatedConvModel
+from common_quantization import test_only_eval_fn
```
FYI this import breaks running under pytest because the imported function's name starts with `test_` :p 

I'll send a quick patch to fix
